File: qc-application/src/main/java/com/eager/questioncloud/application/payment/ChargePointPaymentHistoryService.java
package com.eager.questioncloud.application.payment;

import com.eager.questioncloud.common.PagingInformation;
import com.eager.questioncloud.domain.point.ChargePointPayment;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChargePointPaymentHistoryService {
    private final ChargePointPaymentReader chargePointPaymentReader;

    public List<ChargePointPayment> getChargePointPayments(Long userId, PagingInformation pagingInformation) {
        return chargePointPaymentReader.getChargePointPayments(userId, pagingInformation);
    }

    public int countChargePointPayment(Long userId) {
        return chargePointPaymentReader.countChargePointPayment(userId);
    }
}

File: qc-domain/qc-point/qc-point-api/src/main/kotlin/com/eager/questioncloud/point/service/ChargePointPaymentHistoryService.kt
package com.eager.questioncloud.point.service

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.point.domain.ChargePointPayment
import com.eager.questioncloud.point.infrastructure.repository.ChargePointPaymentRepository
import org.springframework.stereotype.Service

@Service
class ChargePointPaymentHistoryService(
    private val chargePointPaymentRepository: ChargePointPaymentRepository
) {
    fun getChargePointPayments(userId: Long, pagingInformation: PagingInformation): List<ChargePointPayment> {
        return chargePointPaymentRepository.getChargePointPayments(userId, pagingInformation)
    }

    fun countChargePointPayment(userId: Long): Int {
        return chargePointPaymentRepository.countByUserId(userId)
    }
}
